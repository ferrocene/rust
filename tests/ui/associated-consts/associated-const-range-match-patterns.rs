//@ revisions: edition2015 edition2018
//@[edition2015]edition:2015
//@[edition2018]edition:2018..
//@run-pass
#![allow(dead_code, unreachable_patterns)]
#![allow(ellipsis_inclusive_range_patterns)]

struct Foo;

trait HasNum {
    const NUM: isize;
}
impl HasNum for Foo {
    const NUM: isize = 1;
}

fn main() {
    #[cfg(edition2015)]
    assert!(match 2 {
        Foo::NUM ... 3 => true,
        _ => false,
    });
    #[cfg(edition2015)]
    assert!(match 0 {
        -1 ... <Foo as HasNum>::NUM => true,
        _ => false,
    });
    #[cfg(edition2015)]
    assert!(match 1 {
        <Foo as HasNum>::NUM ... <Foo>::NUM => true,
        _ => false,
    });

    assert!(match 2 {
        Foo::NUM ..= 3 => true,
        _ => false,
    });
    assert!(match 0 {
        -1 ..= <Foo as HasNum>::NUM => true,
        _ => false,
    });
    assert!(match 1 {
        <Foo as HasNum>::NUM ..= <Foo>::NUM => true,
        _ => false,
    });
}
