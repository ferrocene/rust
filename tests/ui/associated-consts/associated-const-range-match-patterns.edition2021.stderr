error[E0783]: `...` range patterns are deprecated
  --> $DIR/associated-const-range-match-patterns.rs:19:9
   |
LL |         Foo::NUM ... 3 => true,
   |         ^^^^^^^^^---^^
   |                  |
   |                  help: use `..=` for an inclusive range

error[E0783]: `...` range patterns are deprecated
  --> $DIR/associated-const-range-match-patterns.rs:23:9
   |
LL |         -1 ... <Foo as HasNum>::NUM => true,
   |         ^^^---^^^^^^^^^^^^^^^^^^^^^
   |            |
   |            help: use `..=` for an inclusive range

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0783`.
